main.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"album"
+	"db"
+
+	"github.com/jmoiron/sqlx"
+
+	"github.com/gin-gonic/gin"
+
+	_ "github.com/go-sql-driver/mysql"
+)
+
+func main() {
+	sqlDb := db.NewSql()
+	defer sqlDb.Close()
+
+	r := setupRouter(sqlDb)
+	_ = r.Run(":8080")
+}
+
+func setupRouter(sqlDb *sqlx.DB) *gin.Engine {
+	router := gin.Default()
+	albumRepo := album.New(sqlDb)
+
+	//basic auth for user authentication
+
+	adminAuth := gin.BasicAuth(gin.Accounts{
+		"priyanka@gmail.com": "123",
+	})
+
+	adminauthorized := router.Group("/", adminAuth)
+	{
+
+		adminauthorized.POST("/album", albumRepo.Create)
+		adminauthorized.PUT("/album/edit", albumRepo.Update)
+		adminauthorized.DELETE("/album/remove", albumRepo.Delete)
+
+	}
+
+	basicAuth := gin.BasicAuth(gin.Accounts{
+		"priyanka@gmail.com": "123",
+		"anisha@gmail.com":   "1234",
+	})
+
+	basicauthorized := router.Group("/", basicAuth)
```

Comment:
I think `basicAuth` is the first middleware because every user (either admin or user) can read the album. So it's common functionality for all users.
You have to create second middleware, that is for only admin, from `basicAuth` and in which you have to check `if the email is admin's or not`.

Reply:
I have already created two different middleware. basicauth (for admin and user) , adminauth for admin funcationality.